core/basic_authenticator.go

```diff
@@ -19,15 +19,18 @@ import (
 	"net/http"
 )
 
-// The BasicAuthenticator will perform authentication on outbound requests by adding
-// a "Basic" type Authorization header that contains the base64-encoded username and password.
+// BasicAuthenticator is used to add basic authentication information to requests.
+//
+// Basic Authorization will be sent as an Authorization header in the form:
+//
+//        Authorization: Basic <encoded username and password>
+//
 type BasicAuthenticator struct {
-	// [Required] - the basic auth username and password.
-	Username string
-	Password string
+	Username string // User-supplied username for basic auth.
```

Comment:
I think we should avoid in-line comments like this.   tbh, I think the comment that was there before is fine, but I can see where having a comment above each field would provide an IDE hint to the user.